components/core/src/clp/streaming_compression/lzma/Compressor.cpp

```diff
@@ -0,0 +1,260 @@
+        throw OperationFailed(ErrorCode_Unsupported, __FILENAME__, __LINE__);
+    }
+
+    m_compression_stream = LZMA_STREAM_INIT;
+    init_lzma_encoder(&m_compression_stream, compression_level, m_dict_size);
+
+    // No input upon initialization
+    m_compression_stream.next_in = nullptr;
+    m_compression_stream.avail_in = 0;
+
+    // Attach output buffer to LZMA stream
+    m_compression_stream.next_out = m_compressed_stream_block_buffer.data();
+    m_compression_stream.avail_out = m_compressed_stream_block_buffer.size();
+
+    m_compressed_stream_file_writer = &file_writer;
+
+    m_uncompressed_stream_pos = 0;
+}
+
+auto Compressor::close() -> void {
+    if (nullptr == m_compressed_stream_file_writer) {
+        throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
+    }
+
+    if (m_compression_stream.avail_in > 0) {
+        SPDLOG_ERROR("Tried to close LZMA compressor with unprocessed input data.");
+        throw OperationFailed(ErrorCode_Failure, __FILENAME__, __LINE__);
+    }
+
+    flush_lzma(LZMA_FINISH);
+
+    // Deallocates LZMA stream's internal data structures
+    lzma_end(&m_compression_stream);
+
+    // Detach output buffer from LZMA stream
+    m_compression_stream.next_out = nullptr;
+    m_compression_stream.avail_out = 0;
+
+    m_compressed_stream_file_writer = nullptr;
+}
+
+auto Compressor::write(char const* data, size_t data_length) -> void {
+    if (nullptr == m_compressed_stream_file_writer) {
+        throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
+    }
+
+    if (0 == data_length) {
+        // Nothing needs to be done because we do not need to compress anything
+        return;
+    }
+
+    if (nullptr == data) {
+        throw OperationFailed(ErrorCode_BadParam, __FILENAME__, __LINE__);
+    }
+
+    // Attach input data to LZMA stream
+    m_compression_stream.next_in = clp::size_checked_pointer_cast<uint8_t const>(data);
+    m_compression_stream.avail_in = data_length;
+
+    encode_lzma();
+
+    // All input data have been encoded so detach input data
+    m_compression_stream.next_in = nullptr;
+    m_compression_stream.avail_in = 0;
+
+    m_uncompressed_stream_pos += data_length;
+}
+
+auto Compressor::flush() -> void {
+    if (nullptr == m_compressed_stream_file_writer) {
+        throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
+    }
+    flush_lzma(LZMA_SYNC_FLUSH);
+}
+
+auto Compressor::try_get_pos(size_t& pos) const -> ErrorCode {
+    if (nullptr == m_compressed_stream_file_writer) {
+        return ErrorCode_NotInit;
+    }
+
+    pos = m_uncompressed_stream_pos;
+    return ErrorCode_Success;
+}
+
+auto Compressor::encode_lzma() -> void {
+    while (m_compression_stream.avail_in > 0) {
+        // Write output buffer to file if it's full
+        if (0 == m_compression_stream.avail_out) {
+            flush_stream_output_block_buffer();
+        }
+
+        auto const rc = lzma_code(&m_compression_stream, LZMA_RUN);
```

Comment:
nit:
Earlier we use an initializer list for similar code (auto-typed initialized local variable). Both styles are fine, but lets just make it consistent by having it match.